x-pack/packages/kbn-cloud-security-posture/public/src/utils/query_utils.ts

```diff
@@ -38,3 +43,24 @@ export const decodeQuery = <T extends unknown>(search?: string): Partial<T> | un
   if (!risonQuery) return;
   return decodeRison<T>(risonQuery);
 };
+
+export const encodeQueryUrl = (
+  services: Partial<CoreStart> & CoreStart & CspClientPluginStartDeps,
+  filters: Filter[],
+  groupBy?: string[]
+): any => {
+  return encodeQuery({
+    query: services.data.query.queryString.getDefaultQuery(),
+    filters,
+    ...(groupBy && { groupBy }),
+  });
+};
+
+export const queryFilters = (
+  filterParams: NavFilter = {},
+  dataViewId = SECURITY_DEFAULT_DATA_VIEW_ID
```

Comment:
Hey that was added on [this](https://github.com/elastic/kibana/pull/166500) PR

If we don't specify the DataView on the filter when navigating to the findings page, the FilterManager will  fallback to the default in the sorcerer (`logs-*`). This doesn't prevent filtering from working, but it doesn't allow for editing the filter, since the SearchBar component is [configured](https://github.com/opauloh/kibana/blob/febb198a8ab7b4407d72d881b319c84ed73ee5d3/x-pack/plugins/cloud_security_posture/public/pages/configurations/layout/findings_search_bar.tsx#L56) to accept only the Latest Misconfigurations Dataview (or the Latest Vulnerabilities DataView on the Vulnerabilities tab).

![image](https://github.com/user-attachments/assets/2451d195-85b2-49bc-bb5a-af53b7d83baf)


So the fix was to specify the Dataview for each filter, and then the edit filter works:

![image](https://github.com/user-attachments/assets/d1619088-af6c-457f-8040-c10fdd07614a)

Should probably have added it as a comment, which I suggest @animehart to add so we can recall when going over it in the future.

Another solution could be to update the navigation to set the sorcerer index pattern instead of setting it per filter, but that also involves setting a Dataview in the navigation logic.

Reply:
Thanks for the detailed clarification, makes sense! Agree that we need to add a comment as this Data View logic pieces are quite complicated to grasp without all the context. I think this logic should also be added to the vulnerability navigation logic, right now it looks like it is not part of it, unless I missed that in the code

Reply:
@maxcold 
if there aren't anything that can be broken by it then I think we should

Reply:
as we plan to backport this to 8.16 let's not add this new logic, just to be on the safe side. We can do it separately for 8.17